platformio/src/client_utils.cpp

```diff
@@ -57,33 +58,57 @@
 wl_status_t startWiFi(int &wifiRSSI)
 {
   WiFi.mode(WIFI_STA);
-  Serial.printf("Connecting to '%s'", WIFI_SSID);
-  WiFi.begin(WIFI_SSID, WIFI_PASSWORD);
 
-  // timeout if WiFi does not connect in 10s from now
-  unsigned long timeout = millis() + 10000;
-  wl_status_t connection_status = WiFi.status();
+  WiFiManager wifi_manager;
+  wifi_manager.setConnectTimeout(10);
 
-  while ((connection_status != WL_CONNECTED) && (millis() < timeout))
+  // Turn off config AP fallback to save on battery on random errors
```

Comment:
This code just tries to connect to any previously configured network.  If it fails, the library will go into the AP config mode by default.  I figured this could be bad for battery life so that fallback is disabled on line 66.